gateway-ha/src/main/java/io/trino/gateway/ha/router/TrinoQueryProperties.java

```diff
@@ -114,7 +115,8 @@ public TrinoQueryProperties(
             @JsonProperty("schemas") Set<String> schemas,
             @JsonProperty("catalogSchemas") Set<String> catalogSchemas,
             @JsonProperty("isNewQuerySubmission") boolean isNewQuerySubmission,
-            @JsonProperty("isQueryParsingSuccessful") boolean isQueryParsingSuccessful)
+            @JsonProperty("isQueryParsingSuccessful") boolean isQueryParsingSuccessful,
+            @JsonProperty("parseErrorMessage") Optional<String> parseErrorMessage)
```

Comment:
Why do we need to include "parse" in the field name? Should we add another variable if we find different error cause?

Reply:
Making it a more general field as `errorMessage` and storing any sort of errors here sounds good.